analyses/building-trajectories-monocle/01-building-trajectories-monocle.Rmd

```diff
@@ -0,0 +1,378 @@
+# Find barcodes for lineage1 cells
+lineage1_cells <- colnames(lineage1.cds)[colData(lineage1.cds)$assigned_celltype == lineage1_value_vector]
+# Use the first lineage1 cell as the root (or a vector of root cells)
+lineage1.cds <- order_cells(lineage1.cds, reduction_method = "UMAP", root_cells = lineage1_cells[1])
+
+# Find barcodes for lineage2 cells
+lineage2_cells <- colnames(lineage2.cds)[colData(lineage2.cds)$assigned_celltype == lineage2_value_vector]
+# Use the first lineage2 cell as the root (or a vector of root cells)
+lineage2.cds <- order_cells(lineage2.cds, reduction_method = "UMAP", root_cells = lineage2_cells[1])
+
+
+# Plot trajectories colored by pseudotime
+name <- paste0(plots_dir, "/", glue::glue("Plot-trajectory-lineage1.png"), sep = "")
+print(plot_cells(cds = lineage1.cds, color_cells_by = "pseudotime", show_trajectory_graph = TRUE))
+ggsave(file = name, width = 6, height = 5, device = "png")
+
+name <- paste0(plots_dir, "/", glue::glue("Plot-trajectory-lineage2.png"), sep = "")
+print(plot_cells(cds = lineage2.cds, color_cells_by = "pseudotime", show_trajectory_graph = TRUE))
+ggsave(file = name, width = 6, height = 5, device = "png")
+```
+
+# How to Programmatically Root Cells in a Trajectory Analysis
+
+For exploratory trajectory analysis, it is often useful to root cells based on a specific cluster or cell type. In this example, we root the trajectory using cells from a cluster defined by the `predicted.id` column in the metadata, with the lineage of interest specified by the `r lineage_value` as defined in parameters.
+
+For more information, see the [Monocle3 issue](https://github.com/cole-trapnell-lab/monocle3/issues/328).
+
+```{r programmatically-root-cells, echo=TRUE}
+cds <- as.cell_data_set(seurat_obj) # Convert your Seurat object to a Monocle3 cell_data_set.
+cds <- cluster_cells(cds = cds, reduction_method = "UMAP") # Cluster cells and learn the principal graph using Monocle3.
+cds <- learn_graph(cds, use_partition = TRUE) # to choose more than one node
+#cds <- learn_graph(cds, use_partition = FALSE) # to choose only one node
+
+# Define a function to programmatically select the root node:
+# The function identifies cells with the specified lineage.
+# It finds the principal graph node most frequently associated with these cells.
+# This node is used as the root for ordering cells in pseudotime.
+get_earliest_principal_node <- function(cds, predicted.id = lineage_value){
+  cell_ids <- which(colData(cds)[, "predicted.id"] == lineage_value)
+
+  closest_vertex <-
+  cds@principal_graph_aux[["UMAP"]]$pr_graph_cell_proj_closest_vertex
+  closest_vertex <- as.matrix(closest_vertex[colnames(cds), ])
+  root_pr_nodes <-
+    igraph::V(principal_graph(cds)[["UMAP"]])$name[as.numeric(names
+   (which.max(table(closest_vertex[cell_ids,]))))]
+
+  root_pr_nodes
+}
+
+# Order cells in the trajectory with the selected root node.
+cds <- order_cells(cds, root_pr_nodes=get_earliest_principal_node(cds))
+```
+
```

Comment:
I think it would be useful to include the trajectories on the UMAP plot with the `plot_cells()` function similar to plotting done with the lineages approach above this section.